lib/gssapi/krb5/8003.c

```diff
@@ -163,16 +163,61 @@ _gsskrb5_create_8003_checksum (
     return GSS_S_COMPLETE;
 }
 
+static krb5_error_code
+check_ap_options_cbt(void *ad_data, size_t ad_len,
+                     krb5_boolean *client_asserted_cb)
+{
+    uint32_t ad_ap_options;
+
+    *client_asserted_cb = FALSE;
+
+    if (ad_len != sizeof(uint32_t))
+	return KRB5KRB_AP_ERR_MSG_TYPE;
+
+    _gss_mg_decode_le_uint32(ad_data, &ad_ap_options);
+
+    if (ad_ap_options & KERB_AP_OPTIONS_CBT)
```

Comment:
Should we take this time to also use this to provide integrity protection to all other ap-options?

Reply:
Well, no, MIT isn't putting in the other flags. Missed opportunity. Maybe. I guess we'll add any ap-options that need integrity protection when we add any.